webapp/src/components/StackedChartReport/index.js

```diff
@@ -0,0 +1,186 @@
+import React, { memo, useCallback } from 'react'
+import FileSaver from 'file-saver'
+import PropTypes from 'prop-types'
+import { useTranslation } from 'react-i18next'
+import { useCurrentPng } from 'recharts-to-png'
+import { makeStyles } from '@mui/styles'
+import { Box } from '@mui/system'
+import {
+  ComposedChart,
+  Bar,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  Line,
+  ResponsiveContainer
+} from 'recharts'
+
+import { formatWithThousandSeparator } from '../../utils'
+
+import styles from './styles'
+
+const useStyles = makeStyles(styles)
+
+const CustomTooltip = ({ payload = [], label = '', typeCurrency = '' }) => {
+  const { t } = useTranslation('generalForm')
+
+  return (
+    <div>
+      <strong>{label}</strong>
+      {payload &&
+        payload.map((data, i) => (
+          <div key={`${i}-tooltip`}>{`${
+            data.dataKey === 'EXCHANGE_RATE'
+              ? t('exchangeRate')
+              : data.dataKey === `${typeCurrency}_CLAIMED`
+              ? `${t('claimedCat')} `
+              : data.dataKey === `${typeCurrency}_UNCLAIMED`
+              ? `${t('unclaimedCat')} `
+              : data.dataKey === `${typeCurrency}_CATEGORIZED`
+              ? `${t('categorizedCat')} `
+              : data.dataKey === `${typeCurrency}_UNCATEGORIZED` &&
+                t('uncategorizedCat')
+          }: ${formatWithThousandSeparator(
+            data.payload[data.dataKey],
+            2
+          )}`}</div>
+        ))}
+    </div>
+  )
+}
+CustomTooltip.propTypes = {
+  payload: PropTypes.array,
+  label: PropTypes.any,
+  typeCurrency: PropTypes.any
+}
+
+const RenderLegend = ({ payload, typeCurrency }) => {
+  const { t } = useTranslation('generalForm')
+
+  const classes = useStyles()
+
+  return (
+    <div className={classes.legendContainer}>
+      {payload.map((data, index) => (
+        <label className={classes.labelContainer} key={`item-${index}`}>
+          <Box
+            width={18}
+            height={18}
+            ml={2}
+            mt={0.5}
+            mr={0.5}
+            bgcolor={data.color}
+          />
+
+          {data.dataKey === 'EXCHANGE_RATE'
+            ? t('exchangeRate')
+            : data.dataKey === `${typeCurrency}_CLAIMED`
+            ? `${t('claimedCat')} `
+            : data.dataKey === `${typeCurrency}_UNCLAIMED`
+            ? `${t('unclaimedCat')} `
+            : data.dataKey === `${typeCurrency}_CATEGORIZED`
+            ? `${t('categorizedCat')} `
+            : data.dataKey === `${typeCurrency}_UNCATEGORIZED` &&
+              t('uncategorizedCat')}
+        </label>
+      ))}
+    </div>
+  )
+}
+
+RenderLegend.propTypes = {
+  payload: PropTypes.array,
+  typeCurrency: PropTypes.string
+}
+
+const StakedChartReport = ({
+  data,
+  firstCategory,
+  secondCategory,
+  typeCurrency,
+  showEosRate
+}) => {
+  const classes = useStyles()
```

Comment:
same here, group const declaration

BTW: It's a good example because you have three const declarations but one of them is a function, so, you must order it like that:

first const declaration
```
 const classes = useStyles()
 const [getStakedPng, { ref: stackedRef }] = useCurrentPng()
 ```
 then, a blank space and the function declaration

```
const handleStakedDownload = useCallback(async () => {
    const png = await getStakedPng()

    if (png) {
      FileSaver.saveAs(png, 'staked-chart.png')
    }
  }, [getStakedPng])
  ```